src/postgres/commands/checkAuthentication.ts

```diff
@@ -22,7 +22,7 @@ export async function checkAuthentication(context: IActionContext, treeItem: Pos
                 await enterPostgresCredentials(context, treeItem.parent);
 
                 // Need to configure firewall only for Azure Subscritption accounts
-            } else if (treeItem.parent.resourceGroup && parsedError.errorType === firewallNotConfiguredErrorType) {
+            } else if (treeItem.parent.resourceGroup && (parsedError.errorType === firewallNotConfiguredErrorType || 'ETIMEDOUT') && !(await treeItem.isFirewallRuleSet(treeItem.parent))) {
```

Comment:
```suggestion
            } else if (treeItem.parent.resourceGroup && (parsedError.errorType === firewallNotConfiguredErrorType || parsedError.errorType === 'ETIMEDOUT') && !(await treeItem.isFirewallRuleSet(treeItem.parent))) {
```

Same issue in `PostgresDatabaseTreeItem`. Would be better if the logic was shared so we don't duplicate problems like this

Reply:
In this case, it might actually better to combine 'ETIMEDOUT' condition and 'isFirewallRuleSet()', as 'isFirewallRuleSet()' not needed if a 'firewallNotConfiguredErrorType' is thrown. 

I am working on cleaning this up and maybe get rid of 'ETIMEDOUT' check completely. I will get that out as separate PR.